apis/logging/v1/clusterlogforwarder_types_test.go

```diff
@@ -151,3 +154,385 @@ var _ = Describe("ClusterLogForwarder", func() {
+					Status:             "True",
+					Reason:             "Operator is available",
+					Message:            "Operator is available",
+					LastTransitionTime: metav1.Time{time.Date(0, 0, 0, 0, 0, 0, 0, time.UTC)},
+				},
+			},
+			Inputs: NamedConditions{
+				"name1": {
+					{
+						Type:               "Ready",
+						Status:             "True",
+						Reason:             "name1 is ready",
+						Message:            "name1 is ready with no issue",
+						LastTransitionTime: metav1.Time{time.Date(0, 0, 0, 0, 0, 0, 0, time.UTC)},
+					},
+					{
+						Type:               "Available",
+						Status:             "True",
+						Reason:             "name1 is available",
+						Message:            "name1 is available with no issue",
+						LastTransitionTime: metav1.Time{time.Date(0, 0, 0, 0, 0, 0, 0, time.UTC)},
+					},
+					{
+						Type:               "Error",
+						Status:             "False",
+						Reason:             "name1 has no error",
+						Message:            "name1 has no error due to an issue",
+						LastTransitionTime: metav1.Time{time.Date(0, 0, 0, 0, 0, 0, 0, time.UTC)},
+					},
+				},
+				"name3": {
+					{
+						Type:               "Foo",
+						Status:             "True",
+						Reason:             "Bar",
+						Message:            "Bar's message",
+						LastTransitionTime: metav1.Time{time.Date(2021, 1, 1, 12, 30, 30, 100, time.UTC)},
+					},
+					{
+						Type:               "Foo2",
+						Status:             "False",
+						Reason:             "Bar2",
+						Message:            "Bar2's message",
+						LastTransitionTime: metav1.Time{time.Date(0, 0, 0, 0, 0, 0, 0, time.UTC)},
+					},
+					{
+						Type:               "Error",
+						Status:             "True",
+						Reason:             "name3 has an error",
+						Message:            "name3 has an error due to an issue",
+						LastTransitionTime: metav1.Time{time.Date(0, 0, 0, 0, 0, 0, 0, time.UTC)},
+					},
+				},
+				"name4": {
+					{
+						Type:               "Ready",
+						Status:             "True",
+						Reason:             "name4 is ready",
+						Message:            "name4 is ready with no issue",
+						LastTransitionTime: metav1.Time{time.Date(0, 0, 0, 0, 0, 0, 0, time.UTC)},
+					},
+					{
+						Type:               "Available",
+						Status:             "True",
+						Reason:             "name4 is available",
+						Message:            "name4 is available with no issue",
+						LastTransitionTime: metav1.Time{time.Date(0, 0, 0, 0, 0, 0, 0, time.UTC)},
+					},
+				},
+			},
+		}
+
+		original.Synchronize(&syncTarget)
+		Expect(conditionEquals(original.Conditions, expected.Conditions)).NotTo(HaveOccurred())
+		Expect(namedConditionEquals(original.Inputs, expected.Inputs)).NotTo(HaveOccurred())
+	})
+})
+
+// namedConditionEquals verifies if 2 NamedConditions can be considered the same. It uses conditionEquals to check
+// for equality of conditions.
+func namedConditionEquals(got, expected NamedConditions) error {
```

Comment:
There are condition matchers in /test/matchers.  Please review to see if they cover the needs here or can be adapted accordingly

Reply:
done